include/iocore/dns/SplitDNSYAMLLoader.h

```diff
@@ -0,0 +1,80 @@
+/** @file
+
+  A brief file description
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#pragma once
+
+#include "P_SplitDNSProcessor.h"
+
+#include <swoc/Errata.h>
+#include <swoc/swoc_file.h>
+#include <yaml-cpp/yaml.h>
+
+#include <cstdio>
+#include <ostream>
+#include <string_view>
+#include <system_error>
+#include <utility>
+
+namespace splitdns
+{
+namespace yaml
+{
+  class SplitDNSYAMLLoader
+  {
+  public:
+    static swoc::Errata
```

Comment:
This might be better as `swoc::Rv<self_type>`. Otherwise, what happens to the local loader instance? Doesn't it go out of scope?